rust/exchange_rate/src/lib.rs

```diff
@@ -0,0 +1,251 @@
+use candid::{candid_method, CandidType, Error, Principal};
+use chrono::{DateTime, NaiveDateTime, Utc};
+use ic_cdk;
+use ic_cdk_macros;
+use queues::*;
+use serde::{Deserialize, Serialize};
+use std::cell::RefCell;
+use std::collections::HashMap;
+use tokio;
+use tokio_cron_scheduler::{Job, JobScheduler};
+
+#[derive(CandidType, Serialize, Deserialize, Debug, Clone)]
+pub struct Rate {
+    pub value: f32,
+}
+
+#[derive(CandidType, Clone, Deserialize, Debug, Eq, Hash, PartialEq, Serialize)]
+pub struct HttpHeader {
+    pub name: String,
+    pub value: String,
+}
+
+#[derive(Clone, Debug, PartialEq, CandidType, Eq, Hash, Serialize, Deserialize)]
+pub enum HttpMethod {
+    GET,
+}
+
+#[derive(CandidType, Deserialize, Debug)]
+pub struct CanisterHttpRequestArgs {
+    pub url: String,
+    pub headers: Vec<HttpHeader>,
+    pub body: Option<Vec<u8>>,
+    pub http_method: HttpMethod,
+    pub transform_method_name: Option<String>,
+}
+
+#[derive(CandidType, Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct CanisterHttpResponsePayload {
+    pub status: u64,
+    pub headers: Vec<HttpHeader>,
+    pub body: Vec<u8>,
+}
+
+type Timestamp = i64;
+
+thread_local! {
+    pub static FETCHED: RefCell<HashMap<Timestamp, Rate>>  = RefCell::new(HashMap::new());
+    pub static REQUESTED: RefCell<Queue<Timestamp>> = RefCell::new(Queue::new());
+    pub static SCHEDULER: JobScheduler = JobScheduler::new().unwrap();
```

Comment:
You will have to build a replacement (or find, perhaps the NNS  folks have one) based on https://smartcontracts.org/docs/current/developer-docs/build/languages/motoko/heartbeats/

Reply:
Removed dependency to tokio_job_scheduler, and replaced with IC heartbeat method.